builtins.go

```diff
@@ -131,6 +173,21 @@ func builtinAnd(e *evaluator, xp, yp potentialValue) (value, error) {
 	return y, nil
 }
 
+func builtinOr(e *evaluator, xp, yp potentialValue) (value, error) {
+	x, err := e.evaluateBoolean(xp)
+	if err != nil {
+		return nil, err
+	}
+	if x.value {
+		return x, nil
+	}
+	y, err := e.evaluateBoolean(yp)
```

Comment:
I realize we want to short-circuit, but does this mean we would supress any should-be-static checks of the LHS when the RHS is true?

Reply:
If lhs evaluates to true then we don't evaluate rhs at all. We even have a test for that `or3.input`.

Not sure what you mean by _should-be-static checks_. We statically check for undeclared variables etc. in both arguments.

Reply:
Ah that just shows my lack of understanding of the internals; I had thought a fair amount of static checking was done alongside evaluation.

Reply:
The static checking is limited to 1) not using variables that are not defined 2) not using $, self, super lexically outside of an object.  That is applied to both sides of a || b, however true || error "yolo" will not raise as that's a dynamic error.